geth/transactions/transactor.go

```diff
@@ -0,0 +1,187 @@
+package transactions
+
+import (
+	"context"
+	"math/big"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/status-im/status-go/geth/account"
+	"github.com/status-im/status-go/geth/rpc"
+	"github.com/status-im/status-go/sign"
+)
+
+const (
+	// sendTxTimeout defines how many seconds to wait before returning result in sentTransaction().
+	sendTxTimeout  = 300 * time.Second
+	rpcCallTimeout = time.Minute
+
+	defaultGas = 90000
+)
+
+// Transactor validates, signs transactions.
+// It uses upstream to propagate transactions to the Ethereum network.
+type Transactor struct {
+	signRequests   *sign.PendingRequests
+	upstream       *gethTransactor
```

Comment:
It's upstream to this particular struct, not to the whole node. In-proc RPC is talking to something external, otherwise, we wouldn't use RPC, right? :)
It doesn't really matter if the node happens to run in the same process.
But sure, I'm up for the name suggestions that gives more self-explanatory lines of code than
`t.upstream.SendTransaction` down in this class :)

Reply:
thanks!